ui/src/pages/backoffice/EItem/EItemDetails/components/EItemFiles/EItemFiles.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Loader, Error, ResultsTable } from '@components';
+import _get from 'lodash/get';
+import prettyBytes from 'pretty-bytes';
+import { Icon, Message } from 'semantic-ui-react';
+import { file as fileApi } from '@api';
+import { DownloadButton } from '@pages/backoffice/components/buttons';
+import UploadButton from './UploadButton';
+import { invenioConfig } from '@config';
+import DeleteRecordModal from '@pages/backoffice/components/DeleteRecordModal/DeleteRecordModal';
+
+export default class EItemFiles extends Component {
+  constructor(props) {
+    super(props);
+    this.maxFiles = invenioConfig.eitems.maxFiles;
+  }
+
+  onDelete = row => {
+    this.props.deleteFile(row.bucket, row.key);
+  };
+
+  onSelectFile = file => {
+    if (this.props.files.length >= this.maxFiles) {
+      this.props.addNotification(
```

Comment:
@equadon nit: I am doing also something similar in the [oauth PR](https://github.com/inveniosoftware/invenio-app-ils/pull/638/files#diff-df7011847aeaa5e03ddbfc1cfa49b48cR12) if you like to adopt :)

Reply:
You approach is cleaner, I'll use that too. :)